internal/catalog/workloadselector/mapper.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package workloadselector
+
+import (
+	"context"
+
+	"github.com/hashicorp/consul/internal/controller"
+	"github.com/hashicorp/consul/internal/controller/dependency"
+	"github.com/hashicorp/consul/internal/resource"
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+// MapSelectorToWorkloads will use the "id" index on watched Workload type to find all current
+// workloads selected by the resource.
+func MapSelectorToWorkloads[T WorkloadSelecting](_ context.Context, rt controller.Runtime, r *pbresource.Resource) ([]controller.Request, error) {
+	res, err := resource.Decode[T](r)
+	if err != nil {
+		return nil, err
+	}
+
+	workloads, err := GetWorkloadsWithSelector[T](rt.Cache, res)
+	if err != nil {
+		return nil, err
+	}
+
+	var reqs []controller.Request
```

Comment:
This may be trivial, but it's free to right-size the capacity of this slice here since you know exactly how big it needs to be.

Reply:
I changed it to pre-allocate the slice and then assign elements.